full-stack-tests/backend/src/integration/SchemaSync.test.ts

```diff
@@ -1035,4 +1036,216 @@ describe("Schema synchronization", function (this: Suite) {
+        Arc3d.createXY(Point3d.create(0, 0), 5),
+        Arc3d.createXY(Point3d.create(5, 5), 2),
+        Arc3d.createXY(Point3d.create(-5, -5), 20),
+      ];
+
+      const geometryStream: GeometryStreamProps = [];
+      for (const geom of geomArray) {
+        const arcData = IModelJson.Writer.toIModelJson(geom);
+        geometryStream.push(arcData);
+      }
+
+      const e1 = {
+        classFullName: `TestDomain:Test2dElement`,
+        model: drawingModelId,
+        category: drawingCategoryId,
+        code: Code.createEmpty(),
+        geom: geometryStream,
+        ...args,
+      };
+      return b1.elements.insertElement(e1);;
+    };
+    const updateEl = async (id: Id64String, args: { [key: string]: any }) => {
+      await b1.locks.acquireLocks({ exclusive: id });
+      const updatedElementProps = Object.assign(b1.elements.getElementProps(id), args);
+      b1.elements.updateElement(updatedElementProps);
+    };
+
+    const deleteEl = async (id: Id64String) => {
+      await b1.locks.acquireLocks({ exclusive: id });
+      b1.elements.deleteElement(id);
+    };
+    const getChanges = async () => {
+      return HubMock.downloadChangesets({ iModelId: rwIModelId, targetDir: path.join(KnownTestLocations.outputDir, rwIModelId, "changesets") });
+    };
+
+    const findEl = (id: Id64String, b = b1) => {
+      try {
+        return b.elements.getElementProps(id);
+      } catch (e) {
+        return undefined;
+      }
+    };
+    // 2. Insert a element for the class
+    const el1 = await createEl({ p1: "test1" });
+    const el2 = await createEl({ p1: "test2" });
+    b1.saveChanges();
+    await b1.pushChanges({ description: "insert 2 elements" });
+
+    // 3. Update the element.
+    await updateEl(el1, { p1: "test3" });
+    b1.saveChanges();
+    await b1.pushChanges({ description: "update element 1" });
+
+    // 4. Delete the element.
+    await deleteEl(el2);
+    const el3 = await createEl({ p1: "test4" });
+    b1.saveChanges();
+    await b1.pushChanges({ description: "delete element 2" });
+
+    // 5. import schema and insert element 4 & update element 3
+    await addPropertyAndImportSchema(b1);
+    const el4 = await createEl({ p1: "test5", p2: "test6" });
+    await updateEl(el3, { p1: "test7", p2: "test8" });
+    b1.saveChanges();
+    await b1.pushChanges({ description: "import schema, insert element 4 & update element 3" });
+
+    assert.isDefined(findEl(el1));
+    assert.isUndefined(findEl(el2));
+    assert.isDefined(findEl(el3));
+    assert.isDefined(findEl(el4));
+    assert.deepEqual(Object.getOwnPropertyNames(b1.getMetaData("TestDomain:Test2dElement").properties), ["p1", "p2"]);
+    // 6. Revert to timeline 2
+    await b2.revertAndPushChanges({ toIndex: 3, description: "revert to timeline 2" });
+
+    assert.equal((await getChanges()).at(-1)!.description, "revert to timeline 2");
+    await b1.pullChanges();
+    assert.isUndefined(findEl(el1));
+    assert.isUndefined(findEl(el2));
```

Comment:
[Question] Why el1 and el2 are expected to be undefined in timeline2. Aren't they created in 2nd changeset:

![image](https://github.com/user-attachments/assets/66a39976-66d9-48b2-8989-958e2681b746)

Is `timeline 2` same as `2nd changeset`?

Reply:
they are created in changeset 3. 

* The initialize schema sync push changeset 1
* Setup category push changeset 2
* Insert el1 & el2 make changeset 3